PictureViewMaster/PictureMasterView/PictureMasterImageView.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  PictureMasterImageView.swift
+//  PictureViewMaster
+//
+//  Created by El gera de la gente on 4/6/16.
+//  Copyright © 2016 Inaka. All rights reserved.
+//
+
+import UIKit
+
+@objc protocol MasterViewDelegate: NSObjectProtocol {
+    func showImageInMasterView (image: UIImage)
+}
+
+class PictureMasterImageView: UIImageView {
+    weak var delegate: MasterViewDelegate!
+    
+    override init (frame: CGRect) {
+        super.init(frame: frame)
+        self.addTapGesture()
+    }
+    
+    convenience init () {
+        self.init(frame:CGRect.zero)
+        self.addTapGesture()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.addTapGesture()
+    }
+    
+    convenience init (frame: CGRect, andDelegate delegate: MasterViewDelegate) {
+        self.init(frame: frame)
+        self.delegate = delegate
+        self.addTapGesture()
+    }
+    
+    override init (image: UIImage?) {
+        super.init(image: image)
+        self.addTapGesture()
+    }
+    
+    private func addTapGesture (){
+        self.userInteractionEnabled = true
+        self.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(showImageInMasterView(_:))))
```

Comment:
The `showImage` action is something that's up to the class implementing the delegate. Let's rather remove all that nomenclature from here and just name these methods with what they are here for, which is the detection of the tap.